### PR TITLE
Unhandled exception fixes

### PR DESCRIPTION
**Pull request by Reintjuu**
I tried generated sources from this spec: https://api.swaggerhub.com/apis/Pitane-Mobility/Pitane-API/3.0.1. It gave exceptions in a few cases, so I fixed them.

### CODE REVIEW

**Review by StefH — CHANGES_REQUESTED** [review]
Thank you.

Can you also add this json file as an example like the others?

**Comment by Reintjuu** [comment]
Could you review again?

File: src/RestEaseClientGenerator/Builders/ModelBuilder.cs [hunk 1]
```diff
@@ -85,7 +85,10 @@ public string Build(RestEaseModel restEaseModel, bool isFirst, bool isLast)
                 builder.AppendLine("        /// </summary>");
             }
 
-            var safePropertyName = property.Name.ToValidIdentifier();
+            var safePropertyName = !string.IsNullOrEmpty(property.Name)
+                ? property.Name.ToValidIdentifier()
```
Comment (StefH) [inline src/RestEaseClientGenerator/Builders/ModelBuilder.cs]: Would it make sense to change the implementation from `ToValidIdentifier` to also allow null string as input, and then return string.Empty?

Or will this have other effects for the rest of the code?

**Review by StefH — CHANGES_REQUESTED** [review]
See comments

File: src/RestEaseClientGenerator/Builders/ModelBuilder.cs (hunk 1 above)
Comment (Reintjuu) [reply src/RestEaseClientGenerator/Builders/ModelBuilder.cs]: Yes I think that's better. I'll try to improve it tomorrow.

File: examples/RestEaseClientGeneratorConsoleApp/Examples/Pitane/IPitaneApi.cs [hunk 2]
```diff
@@ -0,0 +1,5046 @@
...
+        [Get("/pitaneCollectTripsByAPI")]
+        Task<object> GetPitaneCollectTripsByAPIAsync([Query(Name = "sub_token")] string subToken, [Query(Name = "pla_datumtijd_from")] string plaDatumtijdFrom, [Query(Name = "pla_datumtijd_to")] string plaDatumtijdTo, [Query(Name = "pla_soortvervoer")] string plaSoortvervoer, [Query(Name = "pla_rolstoel")] bool plaRolstoel, [Query(Name = "pla_rolstoel_excluded")] bool plaRolstoelExcluded, [Query(Name = "pla_eigenaar_filtered")] bool plaEigenaarFiltered, [Query(Name = "pla_eigenaar")] string plaEigenaar);
+
+        /// <summary>
+        /// Create route
+        /// </summary>
+        /// <param name="content">Collection of combined trip records</param>
+        [Post("/pitaneUpdateTripCollection")]
+        [Header("Content-Type", "application/json")]
+        Task<object> PostPitaneUpdateTripCollectionAsync([Body] System.Collections.Generic.List`1[RestEaseClientGenerator.Models.Internal.PropertyDto][] content);
```
Comment (Reintjuu) [inline examples/RestEaseClientGeneratorConsoleApp/Examples/Pitane/IPitaneApi.cs]: Hmm, this looks weird to me. Do you have any idea where this comes from?

File: examples/RestEaseClientGeneratorConsoleApp/Examples/Pitane/IPitaneApi.cs [hunk 3]
```diff
@@ -0,0 +1,5046 @@
...
+
+    /// <summary>
+    /// area in which the asset should be returned as GeoJSON Polygon coordinates
+    /// </summary>
+    public class ConditionReturnAreaReturnArea
+    {
+        /// <summary>
+        /// geojson representation of a polygon. First and last point must be equal. See also https://geojson.org/geojson-spec.html#polygon and example https://geojson.org/geojson-spec.html#id4. The order should be lon, lat [[[lon1, lat1], [lon2,lat2], [lon3,lat3], [lon1,lat1]]], the first point should match the last point.
+        /// </summary>
+        public double[][][]  { get; set; }
```
Comment (Reintjuu) [inline examples/RestEaseClientGeneratorConsoleApp/Examples/Pitane/IPitaneApi.cs]: I noticed we have a name missing here, maybe returning an empty string isn't that good of an idea. Or is this related to something else?

**Review by Reintjuu — COMMENTED** [review]
I found some more issues when looking through the generated client.

File: examples/RestEaseClientGeneratorConsoleApp/Examples/Pitane/IPitaneApi.cs (hunk 2 above)
Comment (StefH) [reply examples/RestEaseClientGeneratorConsoleApp/Examples/Pitane/IPitaneApi.cs]: This is a bug.

I'll check the code.

**Comment by StefH** [comment]
@Reintjuu 
I did some fixes to your branch (I hope you don't mind...). 

So please get the latest.

The only thing which fails now is the 

geojsonPolygon, which is an array of geojsonLine , which is an array of geojsonPoints


I'm stuck on how to fix that.....

File: src/RestEaseClientGenerator/Mappers/ModelsMapper.cs [hunk 4]
```diff
@@ -30,6 +30,21 @@ internal class ModelsMapper : BaseMapper
     {
         foreach (var entry in schemas.OrderBy(s => s.Key))
         {
+            if (entry.Key == "ArrayOfCombinedTrips")
```
Comment (Reintjuu) [inline src/RestEaseClientGenerator/Mappers/ModelsMapper.cs]: I don't quite get these changes, are you defining domain specific logic in the generic mapper? Shouldn't the type of schema be handled in a correct way instead of handling this as an edge case?

File: src/RestEaseClientGenerator/Builders/ModelBuilder.cs (hunk 1 above)
Comment (Reintjuu) [reply src/RestEaseClientGenerator/Builders/ModelBuilder.cs]: I've changed the way ToValidIdentifier works and added an early return, making its input a nullable string.

File: src/RestEaseClientGenerator/Mappers/ModelsMapper.cs (hunk 4 above)
Comment (StefH) [reply src/RestEaseClientGenerator/Mappers/ModelsMapper.cs]: Sorry, this was just some left-over debug code, which can be removed in final version.

**Comment by Reintjuu** [comment]
> @Reintjuu I did some fixes to your branch (I hope you don't mind...).
> 
> So please get the latest.
> 
> The only thing which fails now is the
> 
> geojsonPolygon, which is an array of geojsonLine , which is an array of geojsonPoints
> 
> I'm stuck on how to fix that.....

Wouldn't it be better to just handle it as an array of arrays of arrays? Do we need specific edge case handling for this?

File: src/RestEaseClientGenerator/Mappers/ModelsMapper.cs (hunk 4 above)
Comment (Reintjuu) [reply src/RestEaseClientGenerator/Mappers/ModelsMapper.cs]: Oh, my bad. Could you remove relevant debug code when done?

**Comment by StefH** [comment]
I did a fix.

Can you check?

File: src/RestEaseClientGenerator/Mappers/ModelsMapper.cs (hunk 4 above)
Comment (StefH) [reply src/RestEaseClientGenerator/Mappers/ModelsMapper.cs]: done

File: examples/RestEaseClientGeneratorConsoleApp/Examples/Pitane/IPitaneApi.cs [hunk 5]
```diff
@@ -0,0 +1,5082 @@
...
+        /// <summary>
+        /// the return hours of the facility (if different from operating-hours)
+        /// </summary>
+        public SystemHours[] ReturnHours { get; set; }
+    }
+
+    /// <summary>
+    /// area in which the asset should be returned as GeoJSON Polygon coordinates
+    /// </summary>
+    public class ConditionReturnAreaReturnArea : GeojsonPolygon[]
```
Comment (Reintjuu) [inline examples/RestEaseClientGeneratorConsoleApp/Examples/Pitane/IPitaneApi.cs]: Unfortunately, we're not there yet. We can't have an array as a base type.

**Review by Reintjuu — COMMENTED** [review]
Next to the one comment below. I tried running some of the test configurations in the Program.cs of the Console tester and got some exceptions. I don't know if they ran in the first place, or if we maybe introduced some regression. I suggest to (yes I know, it's better to create a new issue for this) eventually introduce a proper testing setup, with unit/integration tests, using something like [xUnit.net](https://xunit.net/).

**Comment by StefH** [comment]
@Reintjuu 
I applied a hack for now to extend from a List<T> in case extending from an an array.

Note that the json is rather strange, all calls just return "200 ok" without specifying the response object.
Also when I try to add this as a service reference, NSwag cannot handle this json, so maybe there are some other strange definitions.

Can you verify the generated Api + Models, and if possible do some real tests to see if it actually works.

I think I'll merge this PR soon, do some more tests with the microsoft api's.

And indeed, best would be to have some regression tests, however that takes some time to make...

**Comment by Reintjuu** [comment]
Yes, I totally agree. It is a very weird spec. Unfortunately, I can't change its definition, because it's created by a 3rd party. Thanks for all the hard work! Your changes LGTM. I hope we can see this version available on Nuget soon :).

**Review by StefH — APPROVED** [review]

**Comment by StefH** [comment]
I'll create a new NuGet version soon.
